Index: library/eclipseMozilla.c
===================================================================
RCS file: /cvsroot/eclipse/platform-launcher/library/eclipseMozilla.c,v
retrieving revision 1.2
diff -u -r1.2 eclipseMozilla.c
--- library/eclipseMozilla.c	10 Aug 2006 21:21:26 -0000	1.2
+++ library/eclipseMozilla.c	17 Nov 2006 18:39:08 -0000
@@ -102,7 +102,7 @@
 		{
 			struct stat buf;
 			FILE *file = NULL;
-#if defined(__amd64__) || defined(__x86_64__)
+#if defined(__amd64__) || defined(__x86_64__) || defined(__powerpc64__)
 			if (stat("/etc/gre64.conf", &buf) == 0)
 			{
 				file = fopen("/etc/gre64.conf", "r");
