### Eclipse Workspace Patch 1.0
#P org.eclipse.help.base
Index: META-INF/MANIFEST.MF
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.help.base/META-INF/MANIFEST.MF,v
retrieving revision 1.45
diff -u -r1.45 MANIFEST.MF
--- plugins/org.eclipse.help.base/META-INF/MANIFEST.MF	24 Jun 2010 18:38:49 -0000	1.45
+++ plugins/org.eclipse.help.baseMETA-INF/MANIFEST.MF	7 Jul 2010 23:16:48 -0000
@@ -39,8 +39,8 @@
  org.eclipse.help.search,
  org.eclipse.help.server,
  org.eclipse.help.standalone
-Require-Bundle: org.apache.lucene;bundle-version="[1.9.1,2.0.0)";visibility:=reexport,
- org.apache.lucene.analysis;bundle-version="[1.9.1,2.0.0)";visibility:=reexport,
+Require-Bundle: org.apache.lucene;bundle-version="[1.9.1,3.0.0)";visibility:=reexport,
+ org.apache.lucene.analysis;bundle-version="[1.9.1,3.0.0)";visibility:=reexport,
  org.eclipse.ant.core;bundle-version="3.2.200";resolution:=optional,
  org.eclipse.core.runtime;bundle-version="3.6.0",
  org.eclipse.help;bundle-version="[3.5.0,4.0.0)";visibility:=reexport,
