diff -pruN debian.git/build/eclipse-I20090611-1540-fetched-src/features/org.eclipse.equinox.executable/library/gtk/build.sh debian.git.new/build/eclipse-I20090611-1540-fetched-src/features/org.eclipse.equinox.executable/library/gtk/build.sh
--- debian.git/build/eclipse-I20090611-1540-fetched-src/features/org.eclipse.equinox.executable/library/gtk/build.sh	2009-04-23 23:04:23.000000000 +0200
+++ debian.git.new/build/eclipse-I20090611-1540-fetched-src/features/org.eclipse.equinox.executable/library/gtk/build.sh	2009-08-26 23:12:52.882192442 +0200
@@ -153,7 +153,7 @@ PROGRAM_OUTPUT="$programOutput"
 DEFAULT_OS="$defaultOS"
 DEFAULT_OS_ARCH="$defaultOSArch"
 DEFAULT_WS="$defaultWS"
-JAVA_HOME=$javaHome
+
 DEFAULT_JAVA=$defaultJava
 
 LIBRARY_DIR="../../../org.eclipse.equinox.launcher/fragments/org.eclipse.equinox.launcher.$defaultWS.$defaultOS.$defaultOSArch"
diff -pruN debian.git/build/eclipse-I20090611-1540-fetched-src/plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile debian.git.new/build/eclipse-I20090611-1540-fetched-src/plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile
--- debian.git/build/eclipse-I20090611-1540-fetched-src/plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile	2007-12-06 20:30:00.000000000 +0100
+++ debian.git.new/build/eclipse-I20090611-1540-fetched-src/plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile	2009-08-26 23:12:41.594611729 +0200
@@ -17,12 +17,12 @@ LIB_NAME_FULL = liblocalfile_1_0_0.so
 OS_TYPE = linux
 
 #Set this to be the location of your JRE
-JAVA_HOME = ~/vm/sun142
+#JAVA_HOME = ~/vm/sun142
 
 JDK_INCLUDE = -I ${JAVA_HOME}/include -I ${JAVA_HOME}/include/${OS_TYPE}
 COMMON_INCLUDE = -I include
 #OPT_FLAGS=-g
-OPT_FLAGS=-O -s -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+OPT_FLAGS=-O2 -g -s -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 
 core :
 	gcc $(OPT_FLAGS) -fPIC -c $(JDK_INCLUDE) $(COMMON_INCLUDE) -o $(CORE.O) $(CORE.C)
