--- plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF	2012-01-13 09:50:06.794155372 +0100
+++ plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF.old	2012-01-13 09:45:55.000000000 +0100
@@ -5,9 +5,6 @@
 Bundle-Version: 1.0.100.qualifier
 Fragment-Host: org.eclipse.ecf
 Bundle-RequiredExecutionEnvironment: J2SE-1.4
-Import-Package: javax.net,
- javax.net.ssl,
- org.eclipse.osgi.service.security
 Bundle-Localization: plugin
 Export-Package: org.eclipse.ecf.internal.ssl;x-internal:=true
 Bundle-Vendor: %plugin.provider

