--- plugins/org.eclipse.equinox.http.servlet/META-INF/MANIFEST.MF.old	2011-08-17 11:59:30.442517272 -0400
+++ plugins/org.eclipse.equinox.http.servlet/META-INF/MANIFEST.MF	2011-08-17 11:59:55.856808595 -0400
@@ -9,8 +9,8 @@
 Bundle-RequiredExecutionEnvironment: CDC-1.0/Foundation-1.0,
  J2SE-1.3
 Export-Package: org.eclipse.equinox.http.servlet;version="1.1.0"
-Import-Package: javax.servlet;version="2.3",
- javax.servlet.http;version="2.3",
+Import-Package: javax.servlet;version="3.0",
+ javax.servlet.http;version="3.0",
  org.osgi.framework;version="1.3.0",
  org.osgi.service.http;version="[1.2,1.3)"
 Comment-Header: Both Eclipse-LazyStart and Bundle-ActivationPolicy are specified for compatibility with 3.2
--- plugins/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/ServletContextAdaptor.java.old	2011-09-28 15:24:48.659601045 -0400
+++ plugins/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/ServletContextAdaptor.java	2011-09-28 15:25:02.181213633 -0400
@@ -11,12 +11,16 @@
  *******************************************************************************/
 package org.eclipse.equinox.http.servlet.internal;
 
-import java.io.*;
+import java.io.IOException;
+import java.io.InputStream;
 import java.lang.reflect.Method;
 import java.net.URL;
 import java.security.*;
 import java.util.*;
 import javax.servlet.*;
+import javax.servlet.FilterRegistration;
+import javax.servlet.ServletRegistration;
+import javax.servlet.descriptor.JspConfigDescriptor;
 import org.osgi.service.http.HttpContext;
 
 public class ServletContextAdaptor implements ServletContext {
@@ -183,4 +188,134 @@
 		}
 		return null;
 	}
+
+	public FilterRegistration.Dynamic addFilter(String arg0, String arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public FilterRegistration.Dynamic addFilter(String arg0, Filter arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public FilterRegistration.Dynamic addFilter(String arg0, Class arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public void addListener(Class arg0) {
+		// TODO Auto-generated method stub
+
+	}
+
+	public void addListener(String arg0) {
+		// TODO Auto-generated method stub
+
+	}
+
+	public void addListener(EventListener arg0) {
+		// TODO Auto-generated method stub
+
+	}
+
+	public javax.servlet.ServletRegistration.Dynamic addServlet(String arg0, String arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public javax.servlet.ServletRegistration.Dynamic addServlet(String arg0, Servlet arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public javax.servlet.ServletRegistration.Dynamic addServlet(String arg0, Class arg1) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public Filter createFilter(Class arg0) throws ServletException {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public EventListener createListener(Class arg0) throws ServletException {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public Servlet createServlet(Class arg0) throws ServletException {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public void declareRoles(String[] arg0) {
+		// TODO Auto-generated method stub
+
+	}
+
+	public ClassLoader getClassLoader() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public Set getDefaultSessionTrackingModes() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public int getEffectiveMajorVersion() {
+		// TODO Auto-generated method stub
+		return 0;
+	}
+
+	public int getEffectiveMinorVersion() {
+		// TODO Auto-generated method stub
+		return 0;
+	}
+
+	public Set getEffectiveSessionTrackingModes() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public FilterRegistration getFilterRegistration(String arg0) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public Map getFilterRegistrations() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public JspConfigDescriptor getJspConfigDescriptor() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public ServletRegistration getServletRegistration(String arg0) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public Map getServletRegistrations() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public SessionCookieConfig getSessionCookieConfig() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public boolean setInitParameter(String arg0, String arg1) {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	public void setSessionTrackingModes(Set arg0) throws IllegalStateException, IllegalArgumentException {
+		// TODO Auto-generated method stub
+
+	}
 }
