--- plugins/org.eclipse.equinox.p2.director/META-INF/MANIFEST.MF.sav	2011-04-06 10:15:16.444395497 -0500
+++ plugins/org.eclipse.equinox.p2.director/META-INF/MANIFEST.MF	2011-04-06 10:15:31.944114718 -0500
@@ -25,8 +25,8 @@
    org.eclipse.pde.ui",
  org.eclipse.equinox.p2.planner;version="2.0.0"
 Require-Bundle: org.eclipse.equinox.common;bundle-version="[3.3.0,4.0.0)",
- org.sat4j.core;bundle-version="[2.2.0,2.3.0)",
- org.sat4j.pb;bundle-version="[2.2.0,2.3.0)",
+ org.sat4j.core;bundle-version="[2.2.0,3.0.0)",
+ org.sat4j.pb;bundle-version="[2.2.0,3.0.0)",
  org.eclipse.core.jobs;bundle-version="[3.3.0,4.0.0)",
  org.eclipse.equinox.p2.metadata;bundle-version="[2.0.0,3.0.0)"
 Bundle-RequiredExecutionEnvironment: J2SE-1.5,
