### Eclipse Workspace Patch 1.0
#P org.eclipse.swt
Index: Eclipse SWT PI/gtk/library/build.sh
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh,v
retrieving revision 1.54
diff -u -r1.54 build.sh
--- a/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh	29 May 2009 21:30:14 -0000	1.54
+++ b/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh	1 Jun 2009 18:44:43 -0000
@@ -120,19 +120,19 @@
 		MOZILLA_LIBS=`pkg-config --libs mozilla-xpcom`
 		export MOZILLA_INCLUDES
 		export MOZILLA_LIBS
-		MAKE_MOZILLA=make_mozilla
+		MAKE_MOZILLA="make_mozilla make_xpcominit"
 	elif [ x`pkg-config --exists firefox-xpcom && echo YES` = "xYES" ]; then
 		MOZILLA_INCLUDES=`pkg-config --cflags firefox-xpcom`
 		MOZILLA_LIBS=`pkg-config --libs firefox-xpcom`
 		export MOZILLA_INCLUDES
 		export MOZILLA_LIBS
-		MAKE_MOZILLA=make_mozilla
+		MAKE_MOZILLA="make_mozilla make_xpcominit"
 	elif [ x`pkg-config --exists libxul && echo YES` = "xYES" ]; then
-		XULRUNNER_INCLUDES=`pkg-config --cflags libxul`
-		XULRUNNER_LIBS=`pkg-config --libs libxul`
+		XULRUNNER_INCLUDES=`pkg-config --cflags libxul-unstable`
+		XULRUNNER_LIBS="`pkg-config --libs libxul-unstable` -lxpcomglue"
 		export XULRUNNER_INCLUDES
 		export XULRUNNER_LIBS
-		MAKE_MOZILLA=make_xulrunner
+		MAKE_MOZILLA="make_xulrunner make_xpcominit"
 	else
 		echo "None of the following libraries were found:  Mozilla/XPCOM, Firefox/XPCOM, or XULRunner/XPCOM"
 		echo "    *** Mozilla embedding support will not be compiled."
