--- a/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh.orig	2009-12-08 09:00:12.000000000 +0200
+++ b/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh	2010-04-13 17:24:11.418047380 +0300
@@ -83,6 +83,18 @@ if [ "${MODEL}" = "" ]; then
 	fi
 fi
 case $MODEL in
+	arm*)
+		MODEL=arm ;;
+	parisc*)
+		MODEL=PA_RISC ;;
+	sparcv9)
+		MODEL=sparc ;;
+	mips*)
+		if which dpkg-architecture >/dev/null 2>&1; then
+			MODEL=`dpkg-architecture -q-qDEB_HOST_ARCH`
+		fi
+esac
+case $MODEL in
 	"x86_64")
 		SWT_ARCH=x86_64
 		AWT_ARCH=amd64

