Description: Build .xpt file if it does not exit
Author: Benjamin Drung
Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=305569

--- a/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh	
+++ b/plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh	
@@ -24,6 +24,24 @@ if [ "${CC}" = "" ]; then
 	export CC
 fi
 
+COMPONENTS_DIR=$(pwd)/../../org.eclipse.swt/components
+# Check if we have to compile external.xpt from external.idl
+if test ! -f ${COMPONENTS_DIR}/external.xpt; then
+	if test ! -f ${COMPONENTS_DIR}/external.idl; then
+		echo "Can't find ${COMPONENTS_DIR}/external.idl"
+		exit 1
+	fi
+
+	IDLDIR=$(pkg-config --variable=idldir libxul | sed 's@/stable$@@')/unstable
+	if test ! -d ${IDLDIR}; then
+		IDLDIR=$(pkg-config --variable=idldir libxul)
+	fi
+	XPIDL=$(pkg-config --variable=sdkdir libxul)/bin/xpidl
+
+	echo "${XPIDL} -m typelib -I ${IDLDIR} -e ${COMPONENTS_DIR}/external.xpt ${COMPONENTS_DIR}/external.idl"
+	${XPIDL} -m typelib -I ${IDLDIR} -e ${COMPONENTS_DIR}/external.xpt ${COMPONENTS_DIR}/external.idl
+fi
+
 # Determine which OS we are on
 if [ "${OS}" = "" ]; then
 	OS=`uname -s`
