--- a/plugins/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/InternalPlatform.java
+++ b/plugins/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/InternalPlatform.java
@@ -48,6 +48,14 @@
 			Platform.ARCH_X86, //
 			Platform.ARCH_AMD64, // 
 			Platform.ARCH_IA64, //
+			Platform.ARCH_PPC64, //
+			Platform.ARCH_SPARC64, //
+			Platform.ARCH_S390, //
+			Platform.ARCH_S390X, //
+			Platform.ARCH_ALPHA, //
+			Platform.ARCH_ARM, //
+			Platform.ARCH_MIPS, //
+			Platform.ARCH_MIPSEL, //
 			Platform.ARCH_IA64_32};
 
 	// debug support:  set in loadOptions()
--- a/plugins/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Platform.java
+++ b/plugins/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Platform.java
@@ -348,6 +348,15 @@
 	 * @since 3.1
 	 */
 	public static final String ARCH_IA64_32 = "ia64_32";//$NON-NLS-1$
+	
+	public static final String ARCH_PPC64 = "ppc64";//$NON-NLS-1$
+	public static final String ARCH_SPARC64 = "sparc64";//$NON-NLS-1$
+	public static final String ARCH_S390 = "s390";//$NON-NLS-1$
+	public static final String ARCH_S390X = "s390x";//$NON-NLS-1$
+	public static final String ARCH_ALPHA = "alpha";//$NON-NLS-1$
+	public static final String ARCH_ARM = "arm";//$NON-NLS-1$
+	public static final String ARCH_MIPS = "mips";//$NON-NLS-1$
+	public static final String ARCH_MIPSEL = "mipsel";//$NON-NLS-1$
 
 	/**
 	 * Constant string (value "win32") indicating the platform is running on a
--- a/plugins/org.eclipse.osgi/eclipseAdaptor/src/org/eclipse/core/runtime/internal/adaptor/PluginConverterImpl.java
+++ b/plugins/org.eclipse.osgi/eclipseAdaptor/src/org/eclipse/core/runtime/internal/adaptor/PluginConverterImpl.java
@@ -64,7 +64,7 @@
 	private static final String PLUGIN_PROPERTIES_FILENAME = "plugin"; //$NON-NLS-1$
 	private static PluginConverterImpl instance;
 	@SuppressWarnings("deprecation")
-	private static final String[] ARCH_LIST = {org.eclipse.osgi.service.environment.Constants.ARCH_PA_RISC, org.eclipse.osgi.service.environment.Constants.ARCH_PPC, org.eclipse.osgi.service.environment.Constants.ARCH_SPARC, org.eclipse.osgi.service.environment.Constants.ARCH_X86, org.eclipse.osgi.service.environment.Constants.ARCH_AMD64, org.eclipse.osgi.service.environment.Constants.ARCH_IA64};
+	private static final String[] ARCH_LIST = {org.eclipse.osgi.service.environment.Constants.ARCH_PA_RISC, org.eclipse.osgi.service.environment.Constants.ARCH_PPC, org.eclipse.osgi.service.environment.Constants.ARCH_SPARC, org.eclipse.osgi.service.environment.Constants.ARCH_X86, org.eclipse.osgi.service.environment.Constants.ARCH_AMD64, org.eclipse.osgi.service.environment.Constants.ARCH_IA64, org.eclipse.osgi.service.environment.Constants.ARCH_PPC64, org.eclipse.osgi.service.environment.Constants.ARCH_SPARC64, org.eclipse.osgi.service.environment.Constants.ARCH_S390, org.eclipse.osgi.service.environment.Constants.ARCH_S390X, org.eclipse.osgi.service.environment.Constants.ARCH_ALPHA, org.eclipse.osgi.service.environment.Constants.ARCH_ARM, org.eclipse.osgi.service.environment.Constants.ARCH_MIPS, org.eclipse.osgi.service.environment.Constants.ARCH_MIPSEL};
 	static public final String FRAGMENT_MANIFEST = "fragment.xml"; //$NON-NLS-1$
 	static public final String GENERATED_FROM = "Generated-from"; //$NON-NLS-1$
 	static public final String MANIFEST_TYPE_ATTRIBUTE = "type"; //$NON-NLS-1$
--- a/plugins/org.eclipse.osgi/supplement/src/org/eclipse/osgi/service/environment/Constants.java
+++ b/plugins/org.eclipse.osgi/supplement/src/org/eclipse/osgi/service/environment/Constants.java
@@ -160,6 +160,14 @@
 	 */
 	public static final String ARCH_IA64_32 = "ia64_32";//$NON-NLS-1$
 
+	public static final String ARCH_SPARC64 = "sparc64";//$NON-NLS-1$
+	public static final String ARCH_S390 = "s390";//$NON-NLS-1$
+	public static final String ARCH_S390X = "s390x";//$NON-NLS-1$
+	public static final String ARCH_ALPHA = "alpha";//$NON-NLS-1$
+	public static final String ARCH_ARM = "arm";//$NON-NLS-1$
+	public static final String ARCH_MIPS = "mips";//$NON-NLS-1$
+	public static final String ARCH_MIPSEL = "mipsel";//$NON-NLS-1$
+
 	/**
 	 * Constant string (value "win32") indicating the platform is running on a
 	 * machine using the Windows windowing system.
