diff --git a/bundles/org.eclipse.core.net/natives/unix/linux/makefile b/bundles/org.eclipse.core.net/natives/unix/linux/makefile
index 0aa0cbe..a6cf350 100644
--- a/plugins/org.eclipse.core.net/natives/unix/linux/makefile
+++ b/plugins/org.eclipse.core.net/natives/unix/linux/makefile
@@ -1,5 +1,5 @@
 #**********************************************************************
-# Copyright (c) 2008, 2010 Oakland Software Incorporated and others.
+# Copyright (c) 2008, 2012 Oakland Software Incorporated and others.
 # All rights reserved. This program and the accompanying materials 
 # are made available under the terms of the Eclipse Public License v1.0
 # which accompanies this distribution, and is available at
@@ -8,23 +8,27 @@
 # Contributors:
 #	Oakland Software Incorporated - initial submission
 #	IBM Corporation - refactoring, bug 245849
+#	Tristan Tarrant - x86_64 support, bug 273072
+#	Red Hat inc. - using pkg config, bug 369215
 #
 #********************************************************************** 
 #
 # makefile for libgnomeproxy-1.0.0.so
-
+#
+# To build for a specific platform environment,
+# please make sure that TARGET_ENVIRONMENT variable is set to -m32 or -m64
 GNOMEPROXY.C = ../gnomeproxy.c
 GNOMEPROXY.O = gnomeproxy.o
 LIB_NAME = gnomeproxy.so
 LIB_NAME_FULL = libgnomeproxy-1.0.0.so
 
-OS_TYPE = linux
+OS_TYPE = linux
 JDK_INCLUDE = -I${JAVA_HOME}/include -I${JAVA_HOME}/include/${OS_TYPE}
-INCLUDE = -I/usr/include/gconf/2/ -I/usr/include/orbit-2.0/ -I/usr/include/glib-2.0/ -I/usr/lib/glib-2.0/include/
-COMPILER_FLAGS = -O0 -fPIC -g3 -Wall -c -fmessage-length=0 -m32
+INCLUDE = `pkg-config --cflags gconf-2.0`
+COMPILER_FLAGS= -O0 -fPIC -g3 -Wall -c -fmessage-length=0 ${TARGET_ENVIRONMENT}
 
-LIBS := -lgconf-2 -lORBit-2 -lgthread-2.0 -lrt -lgobject-2.0 -lglib-2.0
-LINKER_FLAGS= -m32
+LIBS := `pkg-config --libs gconf-2.0`
+LINKER_FLAGS= ${TARGET_ENVIRONMENT}
 
 all: link
 
diff --git a/bundles/org.eclipse.core.net/natives/unix/linux/x86_64/makefile b/bundles/org.eclipse.core.net/natives/unix/linux/x86_64/makefile
deleted file mode 100644
index 407f9c5..0000000
--- a/plugins/org.eclipse.core.net/natives/unix/linux/x86_64/makefile
+++ /dev/null
@@ -1,48 +0,0 @@
-#**********************************************************************
-# Copyright (c) 2008, 2010 Oakland Software Incorporated and others.
-# All rights reserved. This program and the accompanying materials 
-# are made available under the terms of the Eclipse Public License v1.0
-# which accompanies this distribution, and is available at
-# http://www.eclipse.org/legal/epl-v10.html
-#
-# Contributors:
-#	Oakland Software Incorporated - initial submission
-#	IBM Corporation - refactoring, bug 245849
-#	Tristan Tarrant - x86_64 support, bug 273072
-#
-#********************************************************************** 
-#
-# makefile for libgnomeproxy-1.0.0.so
-
-GNOMEPROXY.C = ../../gnomeproxy.c
-GNOMEPROXY.O = gnomeproxy.o
-LIB_NAME = gnomeproxy.so
-LIB_NAME_FULL = libgnomeproxy-1.0.0.so
-
-OS_TYPE = 
-JDK_INCLUDE = -I${JAVA_HOME}/include -I${JAVA_HOME}/include/${OS_TYPE}
-INCLUDE = `pkg-config --cflags gconf-2.0`
-COMPILER_FLAGS= -O0 -fPIC -g3 -Wall -c -fmessage-length=0 -m64
-
-LIBS := `pkg-config --libs gconf-2.0`
-LINKER_FLAGS= -m64
-
-all: link
-
-compile:
-	@echo "Building file: $(GNOMEPROXY.O)"
-	@echo "Invoking: GCC C Compiler"
-	gcc $(INCLUDE) $(JDK_INCLUDE) $(COMPILER_FLAGS) -o $(GNOMEPROXY.O) $(GNOMEPROXY.C)
-	@echo "Finished building: $(GNOMEPROXY.O)"
-	@echo " "
-
-link: compile
-	@echo "Building target: $(LIB_NAME_FULL)"
-	@echo "Invoking: GCC C Linker"
-	gcc $(LINKER_FLAGS) -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(LIBS) $(GNOMEPROXY.O) -lc
-	@echo "Finished building target: $(LIB_NAME_FULL)"
-	@echo " "
-
-clean:
-	-$(RM) $(GNOMEPROXY.O) $(LIB_NAME_FULL)
-	-@echo " "
\ No newline at end of file
