--- plugins/org.junit_4.8.2.v4_8_2_v20110321-1705/META-INF/MANIFEST.MF.orig	2011-04-22 02:01:40.000000000 -0500
+++ plugins/org.junit_4.8.2.v4_8_2_v20110321-1705/META-INF/MANIFEST.MF	2011-05-08 13:29:52.799585063 -0500
@@ -31,18 +31,3 @@
 Bundle-Version: 4.8.2.v4_8_2_v20110321-1705
 Bundle-ManifestVersion: 2
 
-Name: META-INF/eclipse.inf
-SHA1-Digest: u+F8j/GAE8tzrDry9+wT3Cvg81Y=
-
-Name: junit.jar
-SHA1-Digest: /x3AhpTrJIbIn2UHiKGQO/fYwOs=
-
-Name: about_files/cpl-v10.html
-SHA1-Digest: REygXYi+T14Sj1v8XnnKycNjpN4=
-
-Name: about.html
-SHA1-Digest: Myp5wMdsICQoAejNJG1PhgJ3UXY=
-
-Name: plugin.properties
-SHA1-Digest: jmDQY7WzbFbRBx/fmZNGdzlfNyE=
-
