--- src/org/eclipse/pde/ui/tests/target/LocalTargetDefinitionTests.java	2010-06-14 14:35:00.121364830 -0400
+++ src/org/eclipse/pde/ui/tests/target/LocalTargetDefinitionTests.java	2010-06-14 14:27:24.652186747 -0400
@@ -703,9 +703,10 @@
 		expected.add("org.eclipse.jdt.junit4.runtime.source");
 		expected.add("org.eclipse.jdt.launching.source");
 		expected.add("org.eclipse.jdt.ui.source");
-		expected.add("org.junit.source");
-		expected.add("org.junit4.source");
-		expected.add("org.hamcrest.core.source");
+		/* Removed plugins we don't bundle with. Those were:
+		 * org.junit.source", "org.junit4.source" and
+		 * "org.hamcrest.core.source"
+		 */
 		if (Platform.getOS().equals(Platform.OS_MACOSX)) {
 			expected.add("org.eclipse.jdt.launching.macosx.source");
 			expected.add("org.eclipse.jdt.launching.ui.macosx.source");
