diff -r -Nu eclipse-3.8.0-M1-src/features/org.eclipse.help/feature.xml eclipse-3.8.0-M1-src.patched/features/org.eclipse.help/feature.xml
--- eclipse-3.8.0-M1-src/features/org.eclipse.help/feature.xml	2011-03-27 10:30:27.000000000 -0400
+++ eclipse-3.8.0-M1-src.patched/features/org.eclipse.help/feature.xml	2011-08-23 14:01:19.026328809 -0400
@@ -57,13 +57,6 @@
          unpack="false"/>
 
    <plugin
-         id="org.apache.lucene"
-         download-size="0"
-         install-size="0"
-         version="0.0.0"
-         unpack="false"/>
-
-   <plugin
          id="org.apache.lucene.analysis"
          download-size="0"
          install-size="0"
diff -r -Nu eclipse-3.8.0-M1-src/plugins/org.eclipse.help.base/META-INF/MANIFEST.MF eclipse-3.8.0-M1-src.patched/plugins/org.eclipse.help.base/META-INF/MANIFEST.MF
--- eclipse-3.8.0-M1-src/plugins/org.eclipse.help.base/META-INF/MANIFEST.MF	2011-06-21 11:41:09.000000000 -0400
+++ eclipse-3.8.0-M1-src.patched/plugins/org.eclipse.help.base/META-INF/MANIFEST.MF	2011-08-23 11:09:30.399033718 -0400
@@ -39,7 +39,7 @@
  org.eclipse.help.search,
  org.eclipse.help.server,
  org.eclipse.help.standalone
-Require-Bundle: org.apache.lucene;bundle-version="[2.9.0,3.0.0)";visibility:=reexport,
+Require-Bundle: org.apache.lucene.core;bundle-version="[2.9.0,3.0.0)";visibility:=reexport,
  org.apache.lucene.analysis;bundle-version="[2.9.0,3.0.0)";visibility:=reexport,
  org.eclipse.ant.core;bundle-version="3.2.200";resolution:=optional,
  org.eclipse.core.runtime;bundle-version="3.6.0",
diff -r -Nu eclipse-3.8.0-M1-src/plugins/org.eclipse.help.base/src/org/eclipse/help/internal/search/SearchIndex.java eclipse-3.8.0-M1-src.patched/plugins/org.eclipse.help.base/src/org/eclipse/help/internal/search/SearchIndex.java
--- eclipse-3.8.0-M1-src/plugins/org.eclipse.help.base/src/org/eclipse/help/internal/search/SearchIndex.java	2011-02-01 13:36:06.000000000 -0500
+++ eclipse-3.8.0-M1-src.patched/plugins/org.eclipse.help.base/src/org/eclipse/help/internal/search/SearchIndex.java	2011-08-23 13:45:57.668391818 -0400
@@ -106,7 +106,7 @@
 
 	public static final String DEPENDENCIES_KEY_ANALYZER = "analyzer"; //$NON-NLS-1$
 
-	private static final String LUCENE_BUNDLE_ID = "org.apache.lucene"; //$NON-NLS-1$
+	private static final String LUCENE_BUNDLE_ID = "org.apache.lucene.core"; //$NON-NLS-1$
 
 	private static final String FIELD_NAME = "name"; //$NON-NLS-1$
 
