### Eclipse Workspace Patch 1.0
#P org.eclipse.ant.tests.ui
Index: Ant Debug Tests/org/eclipse/ant/tests/ui/debug/PropertyTests.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.ant.tests.ui/Ant Debug Tests/org/eclipse/ant/tests/ui/debug/PropertyTests.java,v
retrieving revision 1.16
diff -u -r1.16 PropertyTests.java
--- Ant Debug Tests/org/eclipse/ant/tests/ui/debug/PropertyTests.java	1 Feb 2011 17:14:14 -0000	1.16
+++ Ant Debug Tests/org/eclipse/ant/tests/ui/debug/PropertyTests.java	30 Jun 2011 19:28:06 -0000
@@ -24,7 +24,7 @@
 
 public class PropertyTests extends AbstractAntDebugTest {
 	
-	private static final String ANT_VERSION = "Apache Ant(TM) version 1.8.2 compiled on December 20 2010";
+	private static final String ANT_VERSION = "Apache Ant(TM) version 1.8.2";
 
     public PropertyTests(String name) {
 		super(name);
@@ -91,7 +91,7 @@
 			AntProperty property= frame.findProperty("ant.home");
 			assertNotNull(property);
 			
-			assertProperty(thread, "ant.version", ANT_VERSION);
+			assertProperty(thread, "ant.version", ANT_VERSION, false);
 			assertProperty(thread, "ant.project.name", "debugEcho");
 		} finally {
 			terminateAndRemove(thread);
@@ -142,11 +142,18 @@
 	}
 
 	private AntStackFrame assertProperty(AntThread thread, String propertyName, String propertyValue) throws DebugException {
+		return assertProperty(thread, propertyName, propertyValue, true);
+	}
+	
+	private AntStackFrame assertProperty(AntThread thread, String propertyName, String propertyValue, boolean checkFullProperty) throws DebugException {
 		AntStackFrame frame = (AntStackFrame)thread.getTopStackFrame();
 		AntProperty property= frame.findProperty(propertyName);
 		assertNotNull("Did not find property: " + propertyName, property);
 		AntValue value= (AntValue) property.getValue();
-		assertEquals("Value of property "+propertyName+ " incorrect", propertyValue, value.getValueString());
+		if (checkFullProperty)
+			assertTrue("Value of property" + propertyName + " incorrect: " + value.getValueString(), propertyValue.equals(value.getValueString()));
+		else
+			assertTrue("Value of property" + propertyName + " incorrect: " + value.getValueString(), value.getValueString().startsWith(propertyValue));
 		return frame;
 	}
 }
\ No newline at end of file
