### Eclipse Workspace Patch 1.0
#P org.eclipse.ant.tests.core
Index: tests/org/eclipse/ant/tests/core/tests/OptionTests.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.ant.tests.core/tests/org/eclipse/ant/tests/core/tests/OptionTests.java,v
retrieving revision 1.71
diff -u -r1.71 OptionTests.java
--- tests/org/eclipse/ant/tests/core/tests/OptionTests.java	1 Feb 2011 17:14:13 -0000	1.71
+++ tests/org/eclipse/ant/tests/core/tests/OptionTests.java	30 Jun 2011 19:30:21 -0000
@@ -27,7 +27,7 @@
 	
 	protected static final String UNKNOWN_ARG = "Unknown argument: ";
 	protected static final String START_OF_HELP = "ant [options] [target [target2 [target3] ...]]";
-	protected static final String VERSION = "Apache Ant(TM) version 1.8.2 compiled on December 20 2010";
+	protected static final String VERSION = "Apache Ant(TM) version 1.8.2";
 	protected static final String PLUGIN_VERSION = "org.apache.ant_1.8.2";
 	 
 	public OptionTests(String name) {
@@ -58,7 +58,7 @@
 	public void testVersion() throws CoreException {
 		run("TestForEcho.xml", new String[]{"-version"});
 		assertTrue("One message should have been logged", AntTestChecker.getDefault().getMessagesLoggedCount() == 1);
-		assertEquals("Version is incorrect: ", VERSION, getLastMessageLogged());
+		assertTrue("Version is incorrect", getLastMessageLogged().startsWith(VERSION));
 	}
 	
 	/**
