--- plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/make_linux.mak.sav	2011-03-07 23:32:06.523294166 +0200
+++ plugins/org.eclipse.swt/Eclipse SWT PI/gtk/library/make_linux.mak	2011-03-07 23:32:41.850888713 +0200
@@ -92,7 +92,7 @@ MOZILLAEXCLUDES = -DNO__1XPCOMGlueShutdo
 XULRUNNEREXCLUDES = -DNO__1NS_1InitXPCOM2
 
 WEBKITCFLAGS = `pkg-config --cflags gtk+-2.0` -I/usr/include/webkit-1.0 -I/usr/include/libsoup-2.4
-WEBKITLIBS = -lwebkit-1.0
+WEBKITLIBS = -lwebkitgtk-1.0
 
 SWT_OBJECTS = swt.o c.o c_stats.o callback.o
 CDE_OBJECTS = swt.o cde.o cde_structs.o cde_stats.o
