--- plugins/org.eclipse.swt/Eclipse SWT WebKit/gtk/library/webkitgtk.h.sav	2011-04-05 16:50:36.553635440 +0300
+++ plugins/org.eclipse.swt/Eclipse SWT WebKit/gtk/library/webkitgtk.h	2011-04-05 16:50:01.343072638 +0300
@@ -24,6 +24,7 @@
 #include <libsoup/soup-cookie-jar.h>
 #include <libsoup/soup-session-feature.h>
 #include <libsoup/soup-uri.h>
+#include <webkit/webkitglobals.h>
 #include <webkit/webkitdownload.h>
 #include <webkit/webkitnetworkrequest.h>
 #include <webkit/webkitsoupauthdialog.h>
